File: samples/client/3_0_3_unit_test/java/src/test/java/org/openapijsonschematools/client/schemas/validation/TypeValidatorTest.java
package org.openapijsonschematools.client.schemas.validation;

import org.checkerframework.checker.nullness.qual.Nullable;
import org.junit.Assert;
import org.junit.Test;
import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;
import org.openapijsonschematools.client.schemas.StringJsonSchema;

import java.util.ArrayList;
import java.util.LinkedHashSet;

public class TypeValidatorTest {
    @SuppressWarnings("nullness")
    private void assertNull(@Nullable Object object) {
        Assert.assertNull(object);
    }

    @Test
    public void testValidateSucceeds() throws ValidationException {
        final TypeValidator validator = new TypeValidator();
        ValidationMetadata validationMetadata = new ValidationMetadata(
                new ArrayList<>(),
                new SchemaConfiguration(new JsonSchemaKeywordFlags.Builder().build()),
                new PathToSchemasMap(),
                new LinkedHashSet<>()
        );
        @Nullable PathToSchemasMap pathToSchemasMap = validator.validate(
                new ValidationData(
                    StringJsonSchema.StringJsonSchema1.getInstance(),
                    "hi",
                    validationMetadata
                )
        );
        assertNull(pathToSchemasMap);
    }

    @Test
    public void testValidateFailsIntIsNotString() {
        final TypeValidator validator = new TypeValidator();
        ValidationMetadata validationMetadata = new ValidationMetadata(
                new ArrayList<>(),
                new SchemaConfiguration(new JsonSchemaKeywordFlags.Builder().build()),
                new PathToSchemasMap(),
                new LinkedHashSet<>()
        );
        Assert.assertThrows(ValidationException.class, () -> validator.validate(
                new ValidationData(
                    StringJsonSchema.StringJsonSchema1.getInstance(),
                    1,
                    validationMetadata
                )
        ));
    }
}

File: samples/client/3_0_3_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/schemas/validation/TypeValidatorTest.kt
package org.openapijsonschematools.client.schemas.validation

import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException
import org.openapijsonschematools.client.schemas.StringJsonSchema

import kotlin.test.Test
import kotlin.test.assertFailsWith
import kotlin.test.assertNull

class TypeValidatorTest {
    @Test
    @Throws(ValidationException::class)
    fun testValidateSucceeds() {
        val validator = TypeValidator()
        val validationMetadata = ValidationMetadata(
            ArrayList(),
            SchemaConfiguration(JsonSchemaKeywordFlags.Builder().build()),
            PathToSchemasMap(),
            LinkedHashSet()
        )
        val pathToSchemasMap = validator.validate(
            ValidationData(
                StringJsonSchema.StringJsonSchema1.getInstance(),
                "hi",
                validationMetadata
            )
        )
        assertNull(pathToSchemasMap)
    }

    @Test
    fun testValidateFailsIntIsNotString() {
        val validator = TypeValidator()
        val validationMetadata = ValidationMetadata(
            ArrayList(),
            SchemaConfiguration(JsonSchemaKeywordFlags.Builder().build()),
            PathToSchemasMap(),
            LinkedHashSet()
        )
        assertFailsWith<ValidationException>(
            block = {
                validator.validate(
                    ValidationData(
                        StringJsonSchema.StringJsonSchema1.getInstance(),
                        1,
                        validationMetadata
                    )
                )
            }
        )
    }
}